### PR TITLE
KAFKA-9562: part 1: ignore exceptions while flushing stores in close(dirty)

### PR DESCRIPTION
**Pull request by vvcephei**
### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

File: streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamThread.java [hunk 1]
```diff
@@ -701,10 +701,13 @@ private void runLoop() {
...
-                        "This implies that this thread missed a rebalance and dropped out of the consumer group. " +
-                        "Will try to rejoin the consumer group. Below is the detailed description of the task:\n{}",
-                    ignoreAndRejoinGroup.migratedTask().id(), ignoreAndRejoinGroup.migratedTask().toString(">"));
+                log.warn("Detected task " + ignoreAndRejoinGroup.migratedTask().id() +
+                             " that got migrated to another thread. This implies that this thread missed" +
+                             " a rebalance and dropped out of the consumer group. Will try to rejoin the" +
+                             " consumer group. Below is the detailed description of the task:\n" +
+                             ignoreAndRejoinGroup.migratedTask().toString(">"),
+                         ignoreAndRejoinGroup
+                         );
```
Comment (vvcephei) [inline streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamThread.java:710]: I added the exception to the log message, so we can see where the TaskMigratedException is actually caused.

File: streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamTask.java [hunk 2]
```diff
@@ -655,7 +655,11 @@ void suspend(final boolean clean,
             }
         } else {
             // In the case of unclean close we still need to make sure all the stores are flushed before closing any
-            super.flushState();
+            try {
+                stateMgr.flush();
+            } catch (final ProcessorStateException e) {
+                // ignore any exceptions while flushing (all stores would have had a chance to flush anyway)
+            }
```
Comment (vvcephei) [inline streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamTask.java:662]: Judging from the stacktrace generated in the prior logging change, this is where we're throwing a TaskMigratedException, even though this block is an "unclean close". Just trying out ignoring the exception to see what happens.

**Review by vvcephei — COMMENTED** [review]
Hey @abbccdda , this is the branch I created to debug the soak tests. FYI.

File: streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamTask.java (hunk 2 above)
Comment (guozhangwang) [reply streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamTask.java:662]: Just to clarify we should only swallow under `dirty` suspension (in trunk it is wrapped in `close`) right? For clean close / suspend if it throws we would capture it and wrap as TaskMigrated still and then handle them by closing the task as dirty (in that second try we would swallow).

Comment (vvcephei) [reply streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamTask.java:662]: That's right

**Comment by vvcephei** [comment]
IIRC, the test failures were unrelated, but...

Retest this, please.

**Comment by vvcephei** [comment]
Ok to test

**Comment by vvcephei** [comment]
Retest this please

**Comment by vvcephei** [comment]
Ok to test

**Comment by vvcephei** [comment]
Not sure why the tests aren't running... I'm running them locally now.

**Comment by vvcephei** [comment]
All (streams) tests passed locally: `BUILD SUCCESSFUL in 8m 30s`

**Review by guozhangwang — APPROVED** [review]
LGTM!

**Comment by vvcephei** [comment]
Thanks @guozhangwang !

**Pull request merged**